src/MCPClient/lib/clientScripts/archivematicaCreateMETS2.py

```diff
@@ -255,20 +258,48 @@ def createDublincoreDMDSecFromDBData(type, id):
     xmlData.append(dc)
     return (dmdSec, ID)
 
-def createMDRefDMDSec(LABEL, itemdirectoryPath, directoryPathSTR):
+def createDSpaceDMDSec(label, dspace_mets_path, directoryPathSTR):
+    """
+    Parse DSpace METS file and create a dmdSecs from the info.
+
+    One dmdSec will contain a mdRef to the DSpace METS file. The other dmdSec
+    will contain Dublin Core metadata with the identifier and parent collection
+    identifier.
+
+    :param str label: LABEL for the mdRef
+    :param str dspace_mets_path: Path to the DSpace METS
+    :param str directoryPathSTR: Relative path to the DSpace METS
+    :return: dict of {<dmdSec ID>: <dmdSec Element>}
+    """
     global globalDmdSecCounter
+    dmdsecs = collections.OrderedDict()
+    root = etree.parse(dspace_mets_path)
+
+    # Create mdRef to DSpace METS file
     globalDmdSecCounter += 1
-    dmdSec = etree.Element(ns.metsBNS + "dmdSec")
-    ID = "dmdSec_" + globalDmdSecCounter.__str__()
-    dmdSec.set("ID", ID)
-    XPTR = "xpointer(id("
-    tree = etree.parse(itemdirectoryPath)
-    root = tree.getroot()
-    for item in root.findall("{http://www.loc.gov/METS/}dmdSec"):
-        XPTR = "%s %s" % (XPTR, item.get("ID"))
-    XPTR = XPTR.replace(" ", "'", 1) + "'))"
-    newChild(dmdSec, ns.metsBNS + "mdRef", text=None, sets=[("LABEL", LABEL), (ns.xlinkBNS +"href", directoryPathSTR), ("MDTYPE", "OTHER"), ("LOCTYPE","OTHER"), ("OTHERLOCTYPE", "SYSTEM"), ("XPTR", XPTR)])
-    return (dmdSec, ID)
+    dmdid = "dmdSec_" + str(globalDmdSecCounter)
+    dmdsec = etree.Element(ns.metsBNS + "dmdSec", ID=dmdid)
+    dmdsecs[dmdid] = dmdsec
+    xptr_dmdids = [i.get('ID') for i in root.findall("{http://www.loc.gov/METS/}dmdSec")]
+    xptr = "xpointer(id('{}'))".format(' '.join(xptr_dmdids))
```

Comment:
This will bail if the dmdSec does not contain an ID attribute. Easy fix is ```i.get('ID', '')```, but do we want to suppress that error?